pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -96,11 +99,43 @@ func initContainerName(lang language) string {
 	return fmt.Sprintf("datadog-lib-%s-init", lang)
 }
 
+// getDeploymentReference returns the OwnerReference if it is of Kind Deployment
+func getDeploymentReference(pod *corev1.Pod) *v1.OwnerReference {
+	ownerReferences := pod.ObjectMeta.OwnerReferences
+
+	for _, ownerRef := range ownerReferences {
+		if ownerRef.Kind == "Deployment" {
+			return &ownerRef
+		}
+	}
+	return nil
+}
+
 func injectAutoInstrumentation(pod *corev1.Pod, _ string, _ dynamic.Interface) error {
 	if pod == nil {
 		return errors.New("cannot inject lib into nil pod")
 	}
 
+	if config.Datadog.GetBool("admission_controller.auto_instrumentation.apm_instrumentation_enabled") {
```

Comment:
You want to ignore `mutate_unlabelled` but the Webhook selector makes it so that without `mutate_unlabelled`, it should not be possible for this code to be called without the POD having the label? Am I missing something?

Reply:
oh, great catch! Indeed it looks like `buildLabelSelectors` will cause the pods without `admission.datadoghq.com/enabled: true` to be ignored by mutating webhooks... I'll propose a fix for this